src/backend/distributed/utils/metadata_cache.c

```diff
@@ -246,6 +246,34 @@ LoadShardInterval(uint64 shardId)
 
 
 /*
+ * LoadShardPlacement returns the, cached, metadata about a shard placement.
+ *
+ * The return value is a copy of the cached ShardPlacement struct and may
+ * therefore be modified and/or freed.
+ */
+ShardPlacement *
+LoadShardPlacement(uint64 shardId, uint64 placementId)
+{
+	List *shardPlacementList = NIL;
+	ListCell *shardPlacementCell = NULL;
+
+	shardPlacementList = ShardPlacementList(shardId);
```

Comment:
It's a bit annoying to copy the entire list in ShardPlacementList, just to return a single element.  Seems better to copy the relevant bits of code from ShardPlacementList() and just copy the relevant element?